src/types/dto-decorator-factories.type.ts

```diff
@@ -1,57 +1,22 @@
 import { DTODecoratorName } from './dto-decorator-name.type';
 import { DTODecoratorOptions } from './dto-decorator-options.type';
-import { PropertyDecoratorFactory } from './property-decorator-factory.type';
-import { Target } from './target.type';
+import {
+    MandatoryPropertyDecoratorFactory,
+    OptionalPropertyDecoratorFactory,
+} from './property-decorator-factory.type';
+
+/* Some decorator names have mandatory options.
+ */
+export type HasMandatoryOptions = 'IsEnum' | 'IsNested';
```

Comment:
Oh interesting, I wouldn't have thought about this kind of trick. Though, it is kind of odd (but I imagine that's a consequence of the amount of indirection between the factory and actual decorator).